lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/cicCri/ContractTest.java

```diff
@@ -149,13 +150,13 @@ void fetchVerifiableCredential_whenCalledAgainstCicCri_retrievesAValidVc(MockSer
                         credential -> {
                             try {
                                 var vc =
-                                        verifiableCredentialJwtValidator.validate(
+                                        verifiableCredentialJwtValidator.parseAndValidate(
                                                 TEST_USER,
-                                                null,
+                                                TEST_CRI_ID,
                                                 credential,
                                                 VerifiableCredentialConstants
                                                         .IDENTITY_CHECK_CREDENTIAL_TYPE,
-                                                ECKey.parse(CRI_RSA_ENCRYPTION_PUBLIC_JWK),
+                                                ECKey.parse(EC_PUBLIC_JWK),
```

Comment:
Don't we validate signatures using the CRI's public key, not private?

(Conversely, we encrypt the payload to the CRI using our private key, not public)

Reply:
You can validate with the private key as if it were the public key

Reply:
The private key blob there may be inaccurately named - it has both the private and public parts in it so can be used for both.

Reply:
It looks like Dan is right - `CRI_SIGNING_PRIVATE_KEY_JWK` contains both the public and private keys
- x and y are the public parts
- d is the private part

You can compare to the key configuration in core-common-infra, where we only have x and y.

(AFAICT we're stubbing out the crypto in the contract tests anyway, so I'm not sure how important this is!)

Reply:
Ah, I should've elaborated in my earlier comment because this is something I already understood.

I think the key does still matter because we don't mock the verifier (in the validateSignature method):
```java
ECDSAVerifier verifier = new ECDSAVerifier(signingKey);
if (!concatSignatureVerifiableCredential.verify(verifier)) {
```

Reply:
Right, so:
- All the unit tests use the test keys defined in `TestFixtures`
- All the contract tests keys defined in each of the contract tests separately

All of these keys are actually identical, but in principle they could be different.


In terms of this change, we should use the contract test's EC signing key, not the shared fixture, as Dan suggests.

The variable name is also potentially surprising, given that you actually need the public key (not the private key) to verify the signature, but this is far out of scope of this review - it's massive enough already 😬

Reply:
The I've replaced the EC_PUBLIC_JWK references with the private JWKs defined in each contract test 👍